### ISSUE TITLE
Running dangerous :show permission when the resource name is passed and mismatch the controller name

### ISSUE THREAD

**Issue by enmanuelm19** [CanCanCommunity/cancancan#791]
### Steps to reproduce

When creating a custom controller and delegating permissions on another resource, the resource is treated as a parent and defaulting the action to `:show` instead of the controller's action.

```ruby
# ability.rb
def initialize(user)
  can :read, Book
  can :update, Book do
    user.is_admin
  end
end
```

```ruby
class BookServices < ApplicationController
  load_and_authorize_resource :book 

  def update
    head :no_content
  end
end
```
A non admin user can perform the `:update` action when it shouldn't.

### Expected behavior
It should delegate the permissions on the resource set in the helper.
- A permission set for the :book should be checked against when setting the name of the resource in the helper

### Actual behavior
It applies the `:show` action permission because is [hard coded](https://github.com/CanCanCommunity/cancancan/blob/be14fdae4d14cff0a92c719b39889571093d82a5/lib/cancan/controller_resource_loader.rb#L95) unless you set the `parent_action:` option

### System configuration
**Rails version**: 5.2.2

**Ruby version**: 2.7.0

**CanCanCan version**: 3.3.0

**Comment by enmanuelm19** [comment]
I have a [PR](https://github.com/CanCanCommunity/cancancan/pull/792) up to fix this.

If run authorization against the `:show` action is actually intended, at least a section in the documentation should be present to warn people that setting the **resource** on the `load_and_authorize_resource` helper without setting the `parent_action:` option and in controller named differently that the resource will cause authorization always check to the `:show` action which is pretty permissive in general.

**Comment by coorasse** [comment]
I agree completely with the point, but I disagree with the fix. I believe the  `load_and_authorize_resource :book` should recognise that :book is actually not a parent.

**Comment by enmanuelm19** [comment]
> I agree completely with the point, but I disagree with the fix. I believe the `load_and_authorize_resource :book` should recognise that :book is actually not a parent.

Ok, let me close the PR with this comment, I'll come back later to have a better fix for this.

**Comment by h4nz2** [comment]
Hi @enmanuelm19 , I've stumbled upon the same issue today and I find this behaviour very dangerous, was there any follow up on this topic? This behaviour is still live in the current version of this gem.
